tt_eager/queue/queue.cpp

```diff
@@ -1,29 +1,42 @@
+
 // SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
 //
 // SPDX-License-Identifier: Apache-2.0
 
 #include "tt_eager/queue/queue.hpp"
 #include "tt_eager/tt_dnn/op_library/operation.hpp"
-#include "tt_metal/impl/dispatch/command_queue.hpp"
+#include "tt_eager/tt_dnn/op_library/run_operation.hpp"
 
 namespace tt::tt_metal
 {
-    void EnqueueHostToDeviceTransfer(Queue&q, Tensor& dst, void* src, size_t transfer_size)
+    void EnqueueHostToDeviceTransfer(CommandQueue& q, Tensor& dst, const void* src, size_t transfer_size)
     {
+        dst.to(q, MemoryConfig{});
     }
 
-    void EnqueueDeviceToHostTransfer(Queue&q, Tensor& src, void* dst, size_t transfer_size, size_t src_offset)
+    void EnqueueHostToDeviceTransfer(CommandQueue& q, Tensor& dst, const void* src, size_t transfer_size, const MemoryConfig& mem_config)
     {
+        dst.to(q, mem_config);
     }
 
+    void EnqueueDeviceToHostTransfer(CommandQueue& q, Tensor& src, void* dst, size_t transfer_size, size_t src_offset)
+    {
+    }
 
-    void QueueRecordEvent(Queue&q, Event&e){}
-    void QueueWaitForEvent(Queue&q, Event&e){}
-    void EventSynchronize(Event&e){}
-    void QueueSynchronize(Queue&q){}
-
+   void QueueSynchronize(CommandQueue& q){
+        Finish(q);
+    }
 
-    void EnqueueOperation(Queue&q, operation::DeviceOperation& devop, const std::vector<Tensor>& input_tensors, const std::vector<Tensor>& output_tensors){}
-    void EnqueueOperation(Queue&q, operation::DeviceOperation& devop, const std::vector<Tensor>& input_tensors, const std::vector<std::optional<Tensor>>& optional_input_tensors, const std::vector<Tensor>& output_tensors){}
+    void QueueRecordEvent(CommandQueue& q, Event&e){}
+    void QueueWaitForEvent(CommandQueue& q, Event&e){}
+    void EventSynchronize(Event&e){}
```

Comment:
@TT-billteng, are there plans to get this functionality implemented?

Reply:
yes in the future, they wanted stubs for now